src/overlays/actors/ovl_En_Rr/z_en_rr.c

```diff
@@ -21,51 +83,775 @@ const ActorInit En_Rr_InitVars = {
+        this->bodySegs[i].rotTarget.x = 6000.0f;
+        this->bodySegs[i].rotTarget.z = 0.0f;
+    }
+    this->actionFunc = EnRr_Reach;
+    Audio_PlayActorSound2(&this->actor, NA_SE_EN_LIKE_UNARI);
+}
+
+void EnRr_SetupNeutral(EnRr* this) {
+    s32 i;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Rr/func_80AE861C.s")
+    this->reachState = 0;
+    this->segMoveRate = 0.0f;
+    this->segPhaseVelTarget = 2500.0f;
+    for (i = 0; i < 5; i++) {
+        this->bodySegs[i].heightTarget = 0.0f;
+        this->bodySegs[i].rotTarget.x = this->bodySegs[i].rotTarget.z = 0.0f;
+        this->bodySegs[i].scaleTarget.x = this->bodySegs[i].scaleTarget.z = 1.0f;
+    }
+    if (this->retreat) {
+        this->actionTimer = 100;
+        this->actionFunc = EnRr_Retreat;
+    } else {
+        this->actionTimer = 60;
+        this->actionFunc = EnRr_Approach;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Rr/func_80AE8644.s")
+void EnRr_SetupGrabPlayer(EnRr* this, Player* player) {
+    s32 i;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Rr/func_80AE8744.s")
+    this->grabTimer = 100;
+    this->actor.flags &= ~1;
+    this->ocTimer = 8;
+    this->hasPlayer = true;
+    this->reachState = 0;
+    this->segMoveRate = this->swallowOffset = this->actor.speedXZ = 0.0f;
+    this->pulseSizeTarget = 0.15f;
+    this->segPhaseVelTarget = 5000.0f;
+    this->wobbleSizeTarget = 512.0f;
+    for (i = 0; i < 5; i++) {
+        this->bodySegs[i].heightTarget = 0.0f;
+        this->bodySegs[i].rotTarget.x = this->bodySegs[i].rotTarget.z = 0.0f;
+        this->bodySegs[i].scaleTarget.x = this->bodySegs[i].scaleTarget.z = 1.0f;
+    }
+    this->actionFunc = EnRr_GrabPlayer;
+    Audio_PlayActorSound2(&this->actor, NA_SE_EN_LIKE_DRINK);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Rr/func_80AE8810.s")
+u8 EnRr_GetMessage(u8 shield, u8 tunic) {
+    u8 messageIndex = 0;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Rr/func_80AE8918.s")
+    if ((shield == PLAYER_SHIELD_DEKU) || (shield == PLAYER_SHIELD_HYLIAN)) {
+        messageIndex = RR_MESSAGE_SHIELD;
+    }
+    if ((tunic == (PLAYER_TUNIC_GORON + 1)) || (tunic == (PLAYER_TUNIC_ZORA + 1))) {
+        messageIndex |= RR_MESSAGE_TUNIC;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Rr/func_80AE8968.s")
+    return messageIndex;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Rr/func_80AE8B78.s")
+void EnRr_SetupReleasePlayer(EnRr* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+    u8 shield;
+    u8 tunic;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Rr/func_80AE8C44.s")
+    this->actor.flags |= 1;
+    this->hasPlayer = false;
+    this->ocTimer = 110;
+    this->segMoveRate = 0.0f;
+    this->segPhaseVelTarget = 2500.0f;
+    this->wobbleSizeTarget = 2048.0f;
+    tunic = PLAYER_TUNIC_KOKIRI;
+    shield = PLAYER_SHIELD_NONE;
```

Comment:
Imo, PLAYER_TUNIC_* and PLAYER_SHIELD_* enums shouldn't be used when dealing with equipment, because it will get pretty confusing. These enums are specifically intended for use when dealing with player fields/functions.